### ISSUE TITLE
How are submenus supposed to work?

### ISSUE THREAD

**Issue by jesobreira** [kusti8/proton-native#157]
The [docs](https://proton-native.js.org/#/component_APIs/menu) are clear:

> The menu must come outside and before the Window for it to take effect. It is made up of Menu.Items. **Menus can be embedded inside eachother to make sub-menus.**

However, this:

```
<Menu label="Tools">
            <Menu.Item type="Preferences" />
            <Menu.Item>Network</Menu.Item>
            <Menu.Item type="Separator" />
            <Menu.Item>Sign Message</Menu.Item>
            <Menu label="Load Transaction">
                      <Menu.Item>From file</Menu.Item>
                      <Menu.Item>From text</Menu.Item>
            </Menu>
</Menu>
```

Causes this:

<img width="180" alt="captura de tela 2018-06-14 as 00 05 06" src="https://user-images.githubusercontent.com/3002249/41389341-a043de14-6f66-11e8-972b-ec2f0747083b.png">

At least on Mac OS X 10.13.2, the "Load Transaction" item is rather created as a main menu, and not as submenu, despite it being inside a `<Menu>` item.

How was this supposed to be done? Perhaps an example on it could be added to the docs.

**Comment by jesobreira** [comment]
I see. Is this a libui limitation? Or it's something I can help with, on proton-native sphere?

**Comment by parro-it** [comment]
> Is this a libui limitation?

Yes, ATM. See also andlabs/libui#321 and parro-it/libui-node#110

**Comment by jesobreira** [comment]
Got it. May I create a PR that fixes the docs about it?

I also find it interesting to note that menus can be created inside `<React.Fragment>` tags, so that users are able to import menus from external files.

**Comment by mischnic** [comment]
> Got it. May I create a PR that fixes the docs about it?

Sure.
